src/conn/api_calc_modify.c

```diff
@@ -0,0 +1,193 @@
+/*-
+ * Copyright (c) 2014-2018 MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+
+#define	WT_CM_BLOCKSIZE 8
+#define	WT_CM_MINMATCH 64
+#define	WT_CM_STARTGAP (WT_CM_BLOCKSIZE / 2)
+
+typedef struct {
+	const uint8_t *s1, *e1; /* Start / end of pre-image. */
+	const uint8_t *s2, *e2; /* Start / end of after-image. */
+
+	size_t maxdiff;
+	int maxentries;
+} WT_CM_STATE;
+
+typedef struct {
+	const uint8_t *m1, *m2;
+	size_t len;
+} WT_CM_MATCH;
+
+/*
+ * __cm_extend --
+ *	Given a potential match size, extend to find the complete match.
+ */
+static void
+__cm_extend(WT_CM_STATE *cms,
+    const uint8_t *m1, const uint8_t *m2, WT_CM_MATCH *match)
+{
+	const uint8_t *p1, *p2;
+
+	/* Step past the end and before the beginning of the matching block. */
+	for (p1 = m1, p2 = m2;
+	    p1 < cms->e1 && p2 < cms->e2 && *p1 == *p2;
+	    p1++, p2++)
+		;
+
+	for (; m1 >= cms->s1 && m2 >= cms->s2 && *m1 == *m2;
+	    m1--, m2--)
+		;
+
+	match->m1 = m1 + 1;
+	match->m2 = m2 + 1;
+	match->len = p1 > m1 ? (size_t)(p1 - m1 - 1) : 0;
```

Comment:
gcc/clang is going to ding you for `x - y - z`

Reply:
I've never seen this warning with any version of gcc or clang, maybe it's from lint or coverity?  Regardless, I've changed the code.